conformance/tests/generics_scoping.py

```diff
@@ -0,0 +1,80 @@
+# Specification: https://typing.readthedocs.io/en/latest/spec/generics.html#scoping-rules-for-type-variables
+
+# TODO: write PEP 695 versions
+
+from typing import TypeVar, Generic, Iterable, assert_type
+
+# > A type variable used in a generic function could be inferred to represent
+# > different types in the same code block.
+T = TypeVar('T')
+
+def fun_1(x: T) -> T: ...  # T here
+def fun_2(x: T) -> T: ...  # and here could be different
+
+assert_type(fun_1(1), int)
+assert_type(fun_2('a'), str)
+
+# > A type variable used in a method of a generic class that coincides
+# > with one of the variables that parameterize this class is always bound
+# > to that variable.
+
+class MyClass(Generic[T]):
+    def meth_1(self, x: T) -> T: ...  # T here
+    def meth_2(self, x: T) -> T: ...  # and here are always the same
+
+a: MyClass[int] = MyClass()
+a.meth_1(1)
```

Comment:
Maybe add an "# OK" comment to this line. I find that it's easier to score the test if there are explicit "OK" and "Type error" comments.